### PR TITLE
[DNM] kv: add `SeekReverse` method for Retriever.

### PR DESCRIPTION
**Pull request by coocood**

### CODE REVIEW

**Comment by coocood** [comment]
@disksing @shenli PTAL

File: kv/buffer_store.go [hunk 1]
```diff
@@ -66,6 +66,19 @@ func (s *BufferStore) Seek(k Key) (Iterator, error) {
...
+func (s *BufferStore) SeekReverse(k Key) (Iterator, error) {
+	bufferIt, err := s.MemBuffer.SeekReverse(k)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	retrieverIt, err := s.r.SeekReverse(k)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	return newUnionIter(bufferIt, retrieverIt), nil
```
Comment (disksing) [inline kv/buffer_store.go]: I think we should update unionIter too. The merge algorithm depends on key comparison.

**Comment by coocood** [comment]
@disksing PTAL

**Comment by shenli** [comment]
LGTM

**Comment by disksing** [comment]
LGTM

**Pull request merged**